frontend/src/queries/nodes/InsightQuery/utils/filtersToQueryNode.test.ts

```diff
@@ -30,3 +66,1130 @@ describe('actionsAndEventsToSeries', () => {
+                    include_event_types: [PathType.Screen, PathType.PageView],
+                    start_point: 'a',
+                    end_point: 'b',
+                    path_groupings: ['c', 'd'],
+                    funnel_paths: FunnelPathType.between,
+                    funnel_filter: { a: 1 },
+                    exclude_events: ['e', 'f'],
+                    step_limit: 1,
+                    path_start_key: 'g',
+                    path_end_key: 'h',
+                    path_dropoff_key: 'i',
+                    path_replacements: true,
+                    local_path_cleaning_filters: [{ alias: 'home' }],
+                    edge_limit: 1,
+                    min_edge_weight: 1,
+                    max_edge_weight: 1,
+                },
+            }
+            expect(result).toEqual(query)
+        })
+    })
+
+    describe('stickiness filter', () => {
+        it('converts all properties', () => {
+            const filters: Partial<StickinessFilterType> = {
+                insight: InsightType.STICKINESS,
+                compare: true,
+                show_legend: true,
+                hidden_legend_keys: { 0: true, 10: true },
+                stickiness_days: 2,
+                shown_as: ShownAsValue.STICKINESS,
+                display: ChartDisplayType.ActionsLineGraph,
+            }
+
+            const result = filtersToQueryNode(filters)
+
+            const query: Partial<StickinessQuery> = {
+                kind: NodeKind.StickinessQuery,
+                stickinessFilter: {
+                    compare: true,
+                    show_legend: true,
+                    hidden_legend_keys: { 0: true, 10: true },
+                    stickiness_days: 2,
+                    shown_as: ShownAsValue.STICKINESS,
+                    display: ChartDisplayType.ActionsLineGraph,
+                },
+            }
+            expect(result).toEqual(query)
+        })
+    })
+
+    describe('lifecycle filter', () => {
+        it('converts all properties', () => {
+            const filters: Partial<LifecycleFilterType> = {
+                insight: InsightType.LIFECYCLE,
+                shown_as: ShownAsValue.LIFECYCLE,
+            }
+
+            const result = filtersToQueryNode(filters)
+
+            const query: Partial<LifecycleQuery> = {
+                kind: NodeKind.LifecycleQuery,
+                lifecycleFilter: {
+                    shown_as: ShownAsValue.LIFECYCLE,
+                },
+            }
+            expect(result).toEqual(query)
+        })
+    })
+
+    describe('example insights', () => {
+        it('converts `New user retention` insight', () => {
+            const filters: Partial<RetentionFilterType> = {
+                insight: InsightType.RETENTION,
+                period: RetentionPeriod.Week,
+                // TODO: why does the original example have a display here?
+                // display: ChartDisplayType.ActionsTable,
```

Comment:
?

Reply:
retention filter has `DisplayDerivedMixin` which defaults to `ActionsTable` if `display` isn't provided

so I guess we're safe to exclude it